doc/templates/command/template.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef DNF5_COMMANDS_TEMPLATE_TEMPLATE_HPP
+#define DNF5_COMMANDS_TEMPLATE_TEMPLATE_HPP
+
+#include "arguments.hpp"
+
+#include <dnf5/context.hpp>
+#include <libdnf/conf/option_bool.hpp>
+
+#include <memory>
+#include <vector>
+
+
+namespace dnf5 {
+
+// This would define the command template with two options:
+// --foo
+// --bar
+class TemplateCommand : public Command {
+public:
+    explicit TemplateCommand(Command & parent) : Command(parent, "template") {}
+    void set_argument_parser() override;
+    void run() override;
+
+private:
+    // There are two ways to specify options for a command
+
+    // First: C++ Pointer
+    // Create a OptionBool pointer for the option you need.
+    //
+    // There is no problem using a pure C++ pointer since the OptionBool class
+    // will take care of moving the pointer ownership to the parser, which will
+    // be in charge of handling the memory deallocation.
+    libdnf::OptionBool * foo_option{nullptr};
```

Comment:
The pointer is only required to keep access to the value after option parser process all options and arguments. May be it would be nice to mentioned when the parser does it.